sklearn/tests/test_calibration.py

```diff
@@ -455,3 +461,143 @@ def test_calibration_attributes(clf, cv):
         classes = LabelBinarizer().fit(y).classes_
         assert_array_equal(calib_clf.classes_, classes)
         assert calib_clf.n_features_in_ == X.shape[1]
+
+
+@pytest.fixture(scope="module")
+def data():
+    return load_iris(return_X_y=True)
+
+
+@pytest.fixture(scope="module")
+def data_binary(data):
+    X, y = data
+    return X[y < 2], y[y < 2]
+
+
+@pytest.mark.parametrize(
+    "estimator_name, expected_label",
+    [(None, "_line1"),
+     ("my_est", "my_est"),
+     ("my_est2", "my_est2")]
+)
+def test_calibration_display_default_labels(pyplot, estimator_name,
+                                            expected_label):
+    prob_true = np.array([0, 1, 1, 0])
+    prob_pred = np.array([0.2, 0.8, 0.8, 0.4])
+    y_prob = np.array([])
+
+    viz = CalibrationDisplay(prob_true, prob_pred, y_prob,
+                             estimator_name=estimator_name)
+    viz.plot()
+    assert viz.line_.get_label() == expected_label
+
+
+def test_plot_calibration_curve_error_non_binary(pyplot, data):
+    X, y = data
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+
+    msg = "Only binary classification is supported."
+    with pytest.raises(ValueError, match=msg):
+        plot_calibration_curve(clf, X, y)
+
+
+def test_plot_calibration_curve_no_predict_proba(pyplot, data_binary):
+    X, y = data_binary
+
+    class MyClassifier(ClassifierMixin):
+        def fit(self, X, y):
+            self.classes_ = [0, 1]
+            return self
+
+    clf = MyClassifier().fit(X, y)
+
+    msg = "Response method 'predict_proba' not defined in"
+    with pytest.raises(ValueError, match=msg):
+        plot_calibration_curve(clf, X, y)
+
+
+def test_plot_calibration_curve_not_fitted(pyplot, data_binary):
+    X, y = data_binary
+    clf = LogisticRegression()
+
+    with pytest.raises(NotFittedError):
+        plot_calibration_curve(clf, X, y)
+
+
+@pytest.mark.parametrize("n_bins", [5, 10])
+@pytest.mark.parametrize("strategy", ["uniform", "quantile"])
+@pytest.mark.parametrize("with_strings", [True, False])
```

Comment:
with_strings doesn't seem to be used anywhere?

Reply:
Yes, left over from when I had the add brier score option, which has a `pos_label` param.